### PR TITLE
SDL: Don't write garbage to output if it is redirected

### PR DESCRIPTION
**Pull request by Cacodemon345**

### CODE REVIEW

**Review by alexey-lysiuk — COMMENTED** [review]
If output is redirected, whole color parsing seems useless. Can it be skipped completely, and not only additions of color codes?

File: src/common/platform/posix/sdl/i_system.cpp [hunk 1]
```diff
@@ -175,6 +175,7 @@ void I_PrintStr(const char *cp)
 {
 	const char * srcp = cp;
 	FString printData = "";
+	bool terminal = isatty(stdout);
```
Comment (alexey-lysiuk) [inline src/common/platform/posix/sdl/i_system.cpp:178]: `isatty()` function operates on file descriptors, not pointers to `FILE`.

**Review by alexey-lysiuk — CHANGES_REQUESTED** [review]
And please fix

**Comment by Cacodemon345** [comment]
This should be cleaner.

File: src/common/platform/posix/sdl/i_system.cpp (hunk 1 above)
Comment (Cacodemon345) [reply src/common/platform/posix/sdl/i_system.cpp]: Fixed.

**Pull request merged**